pkg/operator/ceph/controller/cleanup.go

```diff
@@ -0,0 +1,158 @@
+		Spec: batch.JobSpec{
+			Template: podSpec,
+		},
+	}
+
+	existingJob := &batch.Job{}
+	err := client.Get(ctx, types.NamespacedName{Name: jobName, Namespace: c.resource.GetNamespace()}, existingJob)
+	if err != nil && !k8sError.IsNotFound(err) {
+		logger.Warningf("failed to detect job %s. %+v", job.Name, err)
+	} else if err == nil {
+		if existingJob.Status.Succeeded == 1 {
+			logger.Infof("found previous clean up job %s as succeeded. Status=%+v", job.Name, existingJob.Status)
+			return true, nil
+		}
+		logger.Infof("found previous clean up job %s as not succeeded. Retrying. Status=%+v", job.Name, existingJob.Status)
+		return false, nil
+	}
+
+	err = client.Create(ctx, job)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to create ceph cleanup job %q", jobName)
+	}
+
+	return false, nil
+}
+
+func (c *ResourceCleanup) jobContainer() v1.Container {
+	volumeMounts := []v1.VolumeMount{}
+	envVars := []v1.EnvVar{}
+	if c.cluster.Spec.DataDirHostPath != "" {
+		hostPathVolumeMount := v1.VolumeMount{Name: volumeName, MountPath: c.cluster.Spec.DataDirHostPath}
+		volumeMounts = append(volumeMounts, hostPathVolumeMount)
+		envVars = append(envVars, []v1.EnvVar{
+			{Name: dataDirHostPath, Value: c.cluster.Spec.DataDirHostPath},
+			{Name: "ROOK_LOG_LEVEL", Value: "DEBUG"},
+			{Name: k8sutil.PodNamespaceEnvVar, Value: c.resource.GetNamespace()},
+		}...)
+	}
+	// append all the resource attributes as env variables.
+	for k, v := range c.config {
+		envVars = append(envVars, v1.EnvVar{Name: k, Value: v})
+	}
+	securityContext := PrivilegedContext(true)
+	return v1.Container{
+		Name:            "resource-cleanup",
+		Image:           c.rookImage,
+		SecurityContext: securityContext,
+		VolumeMounts:    volumeMounts,
+		Env:             envVars,
+		Args:            []string{"ceph", "clean", c.resource.GetObjectKind().GroupVersionKind().Kind},
+		Resources:       cephv1.GetCleanupResources(c.cluster.Spec.Resources),
+	}
+}
+
+func (c *ResourceCleanup) jobTemplateSpec() v1.PodTemplateSpec {
+	volumes := []v1.Volume{}
+	hostPathVolume := v1.Volume{Name: volumeName, VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: c.cluster.Spec.DataDirHostPath}}}
+	volumes = append(volumes, hostPathVolume)
+
+	podSpec := v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: CleanupAppName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				c.jobContainer(),
+			},
+			Volumes:            volumes,
+			RestartPolicy:      v1.RestartPolicyOnFailure,
+			PriorityClassName:  cephv1.GetCleanupPriorityClassName(c.cluster.Spec.PriorityClassNames),
+			ServiceAccountName: k8sutil.DefaultServiceAccount,
+		},
+	}
+
+	return podSpec
+}
+
+// IsCleanupRequired returns true if `ceph.rook.io/force-deletion` annotation is available on the resource
+func IsCleanupRequired(annotations map[string]string) bool {
+	if value, found := annotations[RESOURCE_CLEANUP_ANNOTATION]; found {
+		if value == "true" || value == "True" {
```

Comment:
case-insensitive compare?
```suggestion
		if strings.EqualFold(value, "true") {
```